janus_server/src/message.rs

```diff
@@ -188,6 +189,23 @@ impl Decode for TaskId {
     }
 }
 
+impl TaskId {
+    /// Get a reference to the task ID as a byte slice
+    pub(crate) fn as_bytes(&self) -> &[u8] {
```

Comment:
nit: since the internal `[u8; 32]` is marked `pub`, this method isn't really necessary: anyone could just do `&task_id.0` themselves. Of course, this comes at an understandability cost to folks not already familiar with how `TaskId` is laid out -- so `as_bytes` is better, but why have more than one way to do it?

Maybe we should un-`pub` the internal byte arrays for types encoding a single value, like `TaskId`, `AggregationJobId`, etc, and provide an `as_bytes` method as needed? In this proposal we'd leave structures with multiple values alone. (if you agree, no need to do in this PR, I can pick it up next time I touch this file)

Reply:
Caveat: all that follows is subjective. I can't point to authoritative Rust style guides or anything.

I like the tuple struct pattern for newtypes, but I dislike the `.0`s that crop up in code that uses the newtype, because when I encounter that, I get no hints as to either the type of `.0` nor what it semantically represents. To me, a method like `as_bytes` reads more clearly. However we could achieve the same thing the method does by defining `pub struct TaskId { pub bytes: [u8; 32] }` and then accessing `task_id.bytes`.

I agree with you that establishing a consistent style for newtypes and only having one way to do the thing is a good idea. In the interest of reducing the amount of boilerplate we have to write, I vote for using "regular" structs (i.e., structs with labeled fields) for newtypes.

Reply:
One alternative is we could have such structs Deref into a byte slice, which would mean we could take advantage of automatic dereferencing to reduce boilerplate, but it's a matter of taste whether that's too clever.